index.js

```diff
@@ -531,6 +540,70 @@ Command.prototype.allowUnknownOption = function(arg) {
   return this;
 };
 
+/**
+  * Whether to store option values as properties on command object,
+  * or store separately (specify false). In both cases the option values can be accessed using .opts().
+  *
+  * @param {boolean} value
+  * @return {Command} Command for chaining
+  * @api public
+  */
+
+Command.prototype.storeOptionsAsProperties = function(value) {
+  this._storeOptionsAsProperties = (value === undefined) || value;
+  if (this.options.length) {
+    // This is for programmer, not end user.
+    console.error('Commander usage error: call storeOptionsAsProperties before adding options');
+  }
+  return this;
+};
+
+/**
+  * Whether to pass command to action handler,
+  * or just the options (specify false).
+  *
+  * @param {boolean} value
+  * @return {Command} Command for chaining
+  * @api public
+  */
+
+Command.prototype.passCommandToAction = function(value) {
+  this._passCommandToAction = (value === undefined) || value;
+  return this;
+};
+
+/**
+ * Store option value
+ *
+ * @param {String} key
+ * @param {Object} value
+ * @api private
+ */
+
+Command.prototype._setOptionValue = function(key, value) {
+  if (this._storeOptionsAsProperties) {
+    this[key] = value;
+  } else {
+    this._optionValues[key] = value;
+  }
+};
+
+/**
+ * Retrieve option value
+ *
+ * @param {String} key
+ * @return {Object} value
+ * @api private
+ */
+
+Command.prototype._getOptionValue = function(key) {
+  if (this._storeOptionsAsProperties) {
+    return this[key];
+  } else {
```

Comment:
This is a personal preference, but this 'else' can be deleted.

Reply:
Short version: I'll look into this a little, and probably delete the else.

Long version: I do not use an else when it is an early return like for an error, but I sometimes keep the if/else when they are both normal situations. However, I know some people prefer not to have the redundant else and am ok with that. I think I saw a lint option for this recently. I'll have a look and see if I am in the minority in this code, and we could enforce it. :-)

Reply:
There were only two cases, and they were both introduced by me in this pull request! I have changed them both and added a lint rule.

Reply:
Thank you!

Reply:
Thanks-you for the suggestion. 🙂